### ISSUE TITLE
Error on serialize Uint8Array

### ISSUE THREAD

**Issue by oscarotero** [denoland/deno#13848]
I'm testing [this imagick wasm library](https://github.com/lumeland/imagemagick-deno) in a worker and getting an error on serialize the data result.
This is my code:

```ts
import {
  ImageMagick,
  IMagickImage,
  initializeImageMagick,
} from "https://deno.land/x/imagemagick_deno@0.0.10/mod.ts";


await initializeImageMagick(); // make sure to initialize first!

// Load an image
const content = await Deno.readFile("./img.jpg");

ImageMagick.read(content, (image: IMagickImage) => {
  // Transform the image
  image.resize(50, 50);

  // Get the result
  image.write((data: Uint8Array) => {
    console.log(data);

    structuredClone({
      content: data,
    });
  });
});
```

The `console.log(data)` returns `Uint8Array(1048) [...]` so effectively it's an arrayBuffer. But the postMessage generates the following error:

```
➜  deno run -A --no-check test.ts
Uint8Array(1048) [
  255, 216, 255, 224,  0,  16,  74, 70, 73, 70,   0,   1,   1,  1,   1,
   44,   1,  44,   0,  0, 255, 225,  0, 22, 69, 120, 105, 102,  0,   0,
   77,  77,   0,  42,  0,   0,   0,  8,  0,  0,   0,   0,   0,  0, 255,
  219,   0,  67,   0,  4,   3,   3,  3,  3,  2,   4,   3,   3,  3,   4,
    4,   4,   5,   6, 10,   6,   6,  5,  5,  6,  12,   8,   9,  7,  10,
   14,  12,  15,  14, 14,  12,  13, 13, 15, 17,  22,  19,  15, 16,  21,
   17,  13,  13,  19, 26,  19,  21, 23, 24, 25,
  ... 948 more items
]
error: Uncaught (in promise) DataCloneError: Failed to serialize response
    structuredClone({
    ^
    at serializeJsMessageData (deno:ext/web/13_message_port.js:278:13)
    at structuredClone (deno:ext/web/13_message_port.js:340:25)
    at file:///Users/oscarotero/test.ts:20:5
    at MagickImage.write (https://deno.land/x/imagemagick_deno@0.0.10/src/magick-image.ts:1688:12)
    at file:///Users/oscarotero/test.ts:17:9
    at https://deno.land/x/imagemagick_deno@0.0.10/src/image-magick.ts:146:18
    at Function._use (https://deno.land/x/imagemagick_deno@0.0.10/src/magick-image.ts:1742:14)
    at Function.read (https://deno.land/x/imagemagick_deno@0.0.10/src/image-magick.ts:112:17)
    at file:///Users/oscarotero/test.ts:12:13
```

I don't know why happens. A workaround is creating another array with `new Uint8Array(data)`, but maybe there's a bug in the `structureClone` implementation?

**Comment by andreubotella** [comment]
This happens when copying `Uint8Array`s backed by WebAssembly memory:

```js
// A wasm module that exports a single 1024-byte memory.
const wasmModule = new WebAssembly.Module(new Uint8Array([
    0x00, 0x61, 0x73, 0x6D, 0x01, 0x00, 0x00, 0x00,
    0x05, 0x06, 0x01, 0x01, 0x80, 0x08, 0x80, 0x08,
    0x07, 0x07, 0x01, 0x03, 0x6D, 0x65, 0x6D, 0x02,
    0x00
]));
const instance = new WebAssembly.Instance(wasmModule);
const wasmMemoryView = new Uint8Array(instance.exports.mem.buffer);
structuredClone(wasmMemoryView);
```

Chromium throws with a similar error, but this works fine in Firefox and WebKit. As far as I can tell from the specs, this should be allowed, so this seems like a V8 bug.

**Comment by hironichu** [comment]
I randomly encountered that same error yesterday, tho it wasn't always the happening so i didn't pay attention (I believe it's 1.19 related but I cannot confirm)

**Comment by andreubotella** [comment]
> I randomly encountered that same error yesterday, tho it wasn't always the happening so i didn't pay attention (I believe it's 1.19 related but I cannot confirm)

I can reproduce this bug with the code sample I posted above as far back as Deno 1.13 (and not earlier than that because that was when `structuredClone()` was implemented). If you are indeed getting that same error randomly since upgrading to 1.19 when you weren't getting it before, it'd help a lot to have a sample of that code.

**Comment by andreubotella** [comment]
Filed a V8 bug: https://crbug.com/v8/12703

**Comment by andreubotella** [comment]
This has now been fixed upstream and will land in V8 10.2, corresponding to Chrome 102 and Deno 1.22.

**Comment by littledivy** [comment]
Resolved. Cannot reproduce in newer versions.
